### PR TITLE
 Add API endpoint for comments 

### LINKED ISSUE #45: Add API endpoint for comments

### PR DESCRIPTION
**Pull request by KevinDangVN**
Resolve #45

### CODE REVIEW

File: src/mux/comment.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+import { IComment, ModelComment } from '../model/comment';
+import { ModelUser } from '../model/user';
+import { ArticleUuidValidator, CommentUuidValidator, CommentValidator } from '../validators';
+
+// Get all comments at root level in article for a given article's UUID
+Mux.get<IComment>(
+  '/v1/comment/article/:articleUuid',
+  ArticleUuidValidator,
+  async (requestData: IRequestData): Promise<IResponseList<IComment>> => {
+    const offset = requestData.body.offset > 1 ? requestData.body.offset - 1 : 0;
```
Comment (chiro-hiro) [inline src/mux/comment.ts:13]: It shouldn't be `requestData.body.offset - 1` , we need to keep original value.

File: src/mux/comment.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    const limit = requestData.body.limit ? requestData.body.limit : 10;
```
Comment (chiro-hiro) [inline src/mux/comment.ts:14]: Should have a maximum value for body limit.

Btw, you shouldn't access value directly and pagination field need to be verified too.

File: src/validators/index.ts [hunk 3]
```diff
@@ -42,7 +42,48 @@ export const ArticleValidator = new Validator(
   },
 );
 
+export const ArticleUuidValidator = new Validator({
```
Comment (chiro-hiro) [inline src/validators/index.ts:45]: You could reuse, `uuid` validator

File: src/mux/comment.ts (hunk 1 above)
Comment (KevinDangVN) [reply src/mux/comment.ts:13]: Ok, I will fix it

File: src/validators/index.ts (hunk 3 above)
Comment (KevinDangVN) [reply src/validators/index.ts:45]: I think we still need the ArticleUuidValidator with pagination fields add on, don't we?

Comment (chiro-hiro) [reply src/validators/index.ts:45]: We need a method to reuse schema or concat schema.

**Comment by KevinDangVN** [comment]
Resolve #87

**Comment by KevinDangVN** [comment]
Resolve #87

**Comment by KevinDangVN** [comment]
Resolve #86

**Pull request merged**